kolibri/core/content/public_api.py

```diff
@@ -51,12 +52,16 @@ def retrieve(self, request, pk=None):
         :param pk: id parent node
         :return: an object with keys for each content metadata table and a schema_version key
         """
-        try:
-            UUID(pk)
-        except ValueError:
-            return Response(
-                {"error": "Invalid UUID format."}, status=status.HTTP_400_BAD_REQUEST
-            )
+        if pk is None:
```

Comment:
My apologies here, there seems to be a misunderstanding. we would want to have the try catch block as it. But change the get_object function that is [here](https://github.com/learningequality/kolibri/blob/458bd6032e918f3b1405a40f83df5893fb7ffaaa/kolibri/core/content/public_api.py#L75) to [get_object_or_404](https://docs.djangoproject.com/en/5.1/topics/http/shortcuts/#get-object-or-404). What this would do is we then handle both the cases where PK is either None or invalid. This also aligns with what is implemented in studio [here].(https://github.com/learningequality/studio/blob/d8f95b53cf991f9fd166a34408760aaa37713ce6/contentcuration/kolibri_public/import_metadata_view.py#L102).
Hope this makes sense! Feel free to ask further if you have any questions.

Reply:
No the try catch should be as it was earlier checking if the UUID is in the right format or not. The PK none check would be done by get_object_or_404 function.

Reply:
There are two things that the retrieve function should handle:
1. If the pk is in invalid format then we return 400.
2. If it is None then we return 404.

Now in ImportMetadataViewset the 400 part would be done by try catch block that you implemented earlier and then 404 check would be done by get_object_or_404 function. 

for the other case that in the ContentNodeViewset:
1. For the 404 we would explicitly do it using a check like this:

```
   if pk is None:
       raise Http404
```
2. With this we would have another check that with a try catch block catching the 400 invalid error if the pk is invalid.

Reply:
It would also be good to write test for both the cases in both the viewsets. one for 400 and one for 404 if it does not already exist.